docs/signatory-architecture.md

```diff
@@ -0,0 +1,295 @@
+classDef supportingSystem fill:#666,stroke:#0b4884,color:#ffffff
+classDef cloud fill:#089319,stroke:#089319,color:#ffffff
+class User person
+class SIG,POL,SL,ML,CFG container
+class C cloud
+class P,AWS,G,A,L,LS,Y,T,OC supportingSystem
+```
+### 3. Signatory Component Model 
+A sequence diagram is a way to express the behaviour of the underlying code in a software system. It shows the interactions between different system components over time, highlighting how the system behaves and responds to different inputs. Using a sequence diagram, developers can better understand the flow of data and control within the system and identify potential issues or areas for improvement.
+
+The sequence diagram for the Signatory remote signer application shows the behaviour of the underlying code in the system. It highlights how the different components of the system interact with one another over time, showing the flow of data and control within the system. Using a sequence diagram, developers can better understand how the system behaves and identify potential issues or areas for improvement.
+```mermaid
+sequenceDiagram
+autonumber
+actor Client
+participant Signatory
+participant Signature service
+participant Metrics server
+participant Vault
+
+Signatory->>+Signatory: RootCmd
+Signatory->>+Signatory: read config
+Signatory->>+Signatory: Register Vaults (Yubikey/Azure/CloudKMS)
+Signatory->>+Signatory: Initialize Vault (Yubikey/Azure/CloudKMS)
+Signatory->>+Signatory: ServeCmd
+
+Signatory->>+Signature service: Start service
+Signatory->>+Metrics server: Start service
+
+
+Client->>+Signature service: GET /authorized_keys 
+Signature service->>+Client: Send Authorized keys (Authorised list of clients)
+loop
+Client->>+Signature service: Check the list of configured keys 
+end
+alt If Found
+Client->>Signature service: POST /keys/<account> (Signing key)
+Client->>Signature service: Digest string + payload
+Signature service-->>+Client:Show authorized key not found error
+else Else
+Client-->>+Signature service : continue loop
+end
+
+alt If signing-key found in key-cache
+Signature service->>+Signature service: Return key from cache
+else Else
+Signature service->>+Vault: GET /public-key
+Vault->>+Signature service : Send public key (PEM format)
+Signature service->>+Signature service: Encode secret key into PKH(Public Key Hash)
+Signature service->>+Vault: Request signing operation
+end
+
+Vault->>+Signature service : Send Error or Signature
+Signature service->>+Client: Send Error or Signature
+```
+
+### 4. Tezos Signing Component Model
+
+Tezos uses elliptic curve cryptography to manage private/public key addresses, sign data, and check signatures. Signing a transaction involves hashing the operation request, prefixing it with a watermark, and then signing the resulting byte string with the user's secret key. The signature is then appended to the operation request to create a signed transaction, which can be broadcast to the network for confirmation.
```

Comment:
the text description says first hash, then prepend watermark, the diagram has the order of those 2 reversed. please make consistent and correct the ordering of the hash and watermark steps